File: app/src/main/java/org/mifos/mobilewallet/auth/ui/adapters/OtherBankAdapter.java
package org.mifos.mobilewallet.auth.ui.adapters;

import android.content.Context;
import android.graphics.Color;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import org.mifos.mobilewallet.R;
import org.mifos.mobilewallet.auth.domain.model.Bank;

import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by naman on 20/6/17.
 */

public class OtherBankAdapter extends RecyclerView.Adapter<OtherBankAdapter.ViewHolder> {

    private Context context;
    private List<Bank> otherBanks;

    @Inject
    public OtherBankAdapter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.item_other_banks, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.ivPopularBank.setImageDrawable(
                ContextCompat.getDrawable(context, otherBanks.get(position).getImage()));
        holder.tvPopularBank.setText(otherBanks.get(position).getName());

        if (!otherBanks.get(position).getName().contains("RBL")) {
            holder.tvPopularBank.setTextColor(Color.LTGRAY);
        }
    }

    @Override
    public int getItemCount() {
        if (otherBanks != null) {
            return otherBanks.size();
        } else {
            return 0;
        }
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.iv_popular_bank)
        ImageView ivPopularBank;

        @BindView(R.id.tv_popular_bank)
        TextView tvPopularBank;

        public ViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);
        }
    }

    public void setData(List<Bank> banks) {
        this.otherBanks = banks;
        notifyDataSetChanged();
    }

    public Bank getBank(int position) {
        return otherBanks.get(position);
    }
}

File: mifospay/src/main/java/org/mifospay/bank/adapters/OtherBankAdapter.kt
package org.mifospay.bank.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Filter
import android.widget.Filterable
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import org.mifospay.R
import org.mifospay.domain.model.Bank
import java.util.Locale
import javax.inject.Inject

/**
 * Created by naman on 20/6/17.
 */
class OtherBankAdapter @Inject constructor() : RecyclerView.Adapter<OtherBankAdapter.ViewHolder>(),
    Filterable {
    private var context: Context? = null
    private var otherBanks: List<Bank>? = null
    private var filteredBanks: List<Bank>? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(
            R.layout.item_other_banks, parent,
            false
        )
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.ivPopularBank!!.setImageDrawable(
            ContextCompat.getDrawable(context!!, otherBanks!![position].image)
        )
        holder.tvPopularBank!!.text = otherBanks!![position].name
    }

    override fun getItemCount(): Int {
        return if (otherBanks != null) {
            otherBanks!!.size
        } else {
            0
        }
    }

    fun setContext(context: Context?) {
        this.context = context
    }

    fun setData(banks: List<Bank>?) {
        otherBanks = banks
        notifyDataSetChanged()
    }

    fun getBank(position: Int): Bank {
        return otherBanks!![position]
    }

    override fun getFilter(): Filter {
        return object : Filter() {
            override fun performFiltering(constraint: CharSequence): FilterResults {
                val charString = constraint.toString()
                filteredBanks = if (charString.isEmpty()) {
                    otherBanks
                } else {
                    val filteredList: MutableList<Bank> = ArrayList()
                    for (bank in otherBanks!!) {
                        if (bank.name.lowercase(Locale.getDefault())
                                .contains(charString.lowercase(Locale.getDefault()))
                        ) {
                            filteredList.add(bank)
                        }
                    }
                    filteredList
                }
                val filterResults = FilterResults()
                filterResults.values = filteredBanks
                return filterResults
            }

            override fun publishResults(constraint: CharSequence, results: FilterResults) {
                filteredBanks = results.values as List<Bank>
                notifyDataSetChanged()
            }
        }
    }

    fun filterList(filterdNames: List<Bank>?) {
        otherBanks = filterdNames
        notifyDataSetChanged()
    }

    inner class ViewHolder(v: View?) : RecyclerView.ViewHolder(v!!) {
        @JvmField
        @BindView(R.id.iv_popular_bank)
        var ivPopularBank: ImageView? = null

        @JvmField
        @BindView(R.id.tv_popular_bank)
        var tvPopularBank: TextView? = null

        init {
            ButterKnife.bind(this, v!!)
        }
    }
}